src/ui/public/persisted_state/__tests__/persisted_state.js

```diff
@@ -0,0 +1,536 @@
+      persistedState = new PersistedState(val);
+
+      expect(persistedState.get()).to.eql(val);
+      expect(persistedState.get()).to.not.equal(val);
+    });
+
+    it('should not throw if creating valid child object', function () {
+      var run = function () {
+        var val = { red: 'blue' };
+        var path = ['test.path'];
+        var parent = new PersistedState();
+        new PersistedState(val, path, parent);
+      };
+
+      expect(run).not.to.throwException();
+    });
+
+    it('should throw if given an invalid value', function () {
+      var run = function () {
+        var val = 'bananas';
+        new PersistedState(val);
+      };
+
+      expect(run).to.throwException(function (err) {
+        expect(err).to.be.a(errors.PersistedStateError);
+      });
+    });
+
+    it('should not throw if given primitive to child', function () {
+      var run = function () {
+        var val = 'bananas';
+        var path = ['test.path'];
+        var parent = new PersistedState();
+        new PersistedState(val, path, parent);
+      };
+
+      expect(run).not.to.throwException();
+    });
+
+    it('should throw if given an invalid parent object', function () {
+      var run = function () {
+        var val = { red: 'blue' };
+        var path = ['test.path'];
+        var parent = {};
+        new PersistedState(val, path, parent);
+      };
+
+      expect(run).to.throwException(function (err) {
+        expect(err).to.be.a(errors.PersistedStateError);
+      });
+    });
+
+    it('should throw if given a parent without a path', function () {
+      var run = function () {
+        var val = { red: 'blue' };
+        var path;
+        var parent = new PersistedState();
+
+        new PersistedState(val, path, parent);
+      };
+
+      expect(run).to.throwException(function (err) {
+        expect(err).to.be.a(errors.PersistedStateError);
+      });
+    });
+  });
+
+  describe('child state creation', function () {
+    var childState;
+
+    it('should not append the child state to the parent, without parent value', function () {
+      var childIndex = 'i can haz child';
+      var persistedState = new PersistedState();
+      childState = persistedState.createChild(childIndex);
+
+      // parent state should contain the child and its original state
+      expect(persistedState.get()).to.not.have.property(childIndex);
+    });
+
+    it('should not append the child state to the parent, with parent value', function () {
+      var childIndex = 'i can haz child';
+      var persistedStateValue = { original: true };
+      var persistedState = new PersistedState(persistedStateValue);
+      childState = persistedState.createChild(childIndex);
+
+      // child state should be empty, we didn't give it any default data
+      expect(childState.get()).to.be(undefined);
+
+      // parent state should contain the child and its original state value
+      expect(persistedState.get()).to.not.have.property(childIndex);
```

Comment:
Shouldn't this also have some check like, `expect(persistedState.get()).to.eql(persistedStateValue);` if it's supposed to ensure it contains the original state value?